compiler/src/dotty/tools/dotc/core/TypeComparer.scala

```diff
@@ -2588,7 +2588,7 @@ class TrackingTypeComparer(initctx: Context) extends TypeComparer(initctx) {
               val instances = paramInstances(new Array(caseLambda.paramNames.length), pat)
               instantiateParams(instances)(body)
             case _ =>
-              body
+              body.simplified
```

Comment:
Why `simplified` here?

Reply:
The added test case (with `case Int =>` instead of `case _ =>`) fails to compile on master with the following:

```scala
11 |  val fib2: Fib[2] = 1
   |                     ^
   |                  Found:    (1 : Int)
   |                  Required: LazyRef(Test2.Fib[(2 : Int) - (1 : Int)]) + 
   |                    LazyRef(Test2.Fib[(2 : Int) - (2 : Int)])
```

That's because when match type reduction simply reduces to `body` without going though `instantiateParams` reduction would stop without trying to further simplify the `body` in question.